nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -306,114 +298,130 @@ void Optimizer::propagateStateVelocitiesFromInitials(
 }
 
 void Optimizer::integrateStateVelocities(
-  xt::xtensor<float, 2> & trajectory,
-  const xt::xtensor<float, 2> & sequence) const
+  Eigen::Array<float, Eigen::Dynamic, 3> & trajectory,
+  const Eigen::ArrayXXf & sequence) const
 {
   float initial_yaw = static_cast<float>(tf2::getYaw(state_.pose.pose.orientation));
 
-  const auto vx = xt::view(sequence, xt::all(), 0);
-  const auto wz = xt::view(sequence, xt::all(), 1);
+  const auto vx = sequence.col(0);
+  const auto wz = sequence.col(1);
 
-  auto traj_x = xt::view(trajectory, xt::all(), 0);
-  auto traj_y = xt::view(trajectory, xt::all(), 1);
-  auto traj_yaws = xt::view(trajectory, xt::all(), 2);
+  auto traj_x = trajectory.col(0);
+  auto traj_y = trajectory.col(1);
+  auto traj_yaws = trajectory.col(2);
 
-  xt::noalias(traj_yaws) = xt::cumsum(wz * settings_.model_dt, 0) + initial_yaw;
+  size_t n_size = traj_yaws.size();
+
+  traj_yaws(0) = wz(0) * settings_.model_dt + initial_yaw;
+  float last_yaw = traj_yaws(0);
+  for(size_t i = 1; i != n_size; i++) {
+    float & curr_yaw = traj_yaws(i);
+    curr_yaw = last_yaw + wz(i) * settings_.model_dt;
+    last_yaw = curr_yaw;
+  }
 
-  auto yaw_cos = xt::roll(xt::eval(xt::cos(traj_yaws)), 1);
-  auto yaw_sin = xt::roll(xt::eval(xt::sin(traj_yaws)), 1);
-  xt::view(yaw_cos, 0) = cosf(initial_yaw);
-  xt::view(yaw_sin, 0) = sinf(initial_yaw);
+  utils::shiftColumnsByOnePlace(traj_yaws, 1);
```

Comment:
I believe the previous lines like `auto yaw_cos = xt::roll(xt::eval(xt::cos(traj_yaws)), 1);` will do the cosine of the `traj_yaws`, evaluate it into a tensor and roll that tensor in return. That would not then roll the `traj_yaws` itself. 

The lines that we have now look like they modify the `traj_yaws` which isn't correct since that is not a copy of `trajectory.col(2)` but a reference to it that can be modified, yes?

Reply:
Agreed, I missed this.